core/memory.go

```diff
@@ -50,6 +50,7 @@ func (o *MemoryObject) Writer() (ObjectWriter, error) {
 
 func (o *MemoryObject) Write(p []byte) (n int, err error) {
 	o.cont = append(o.cont, p...)
+	o.sz = int64(len(o.cont))
```

Comment:
This maybe breaking the intended use of the setters of this object.

It is difficult to tell as the intended usage is not written anywhere, but if I remember correctly, @mcuadros wants to be able to write the size of the content and the content as to independent things, meaning you can have a content of 10 bytes but a length of 13.

This means this change is a bug and can break how are we using this library from the outside, but I am not sure.

Please @mcuadros, comment on this.